test/arm/neon/abal.c

```diff
@@ -0,0 +1,392 @@
+    { { INT16_C(23148), INT16_C(12510), -INT16_C(9344), INT16_C(9847), INT16_C(5308), INT16_C(16725), INT16_C(25730), -INT16_C(3498) },
+      { INT8_C(19), -INT8_C(122), INT8_C(25), INT8_C(42), -INT8_C(70), -INT8_C(61), -INT8_C(91), -INT8_C(101) },
+      { INT8_C(59), -INT8_C(111), INT8_C(36), INT8_C(1), INT8_C(70), INT8_C(77), INT8_C(51), INT8_C(31) },
+      { INT16_C(23188), INT16_C(12521), -INT16_C(9333), INT16_C(9888), INT16_C(5448), INT16_C(16863), INT16_C(25872), -INT16_C(3366) } },
+    { { INT16_C(688), -INT16_C(20307), -INT16_C(6465), -INT16_C(14204), -INT16_C(18860), -INT16_C(19839), INT16_C(11212), -INT16_C(9807) },
+      { -INT8_C(94), INT8_C(1), -INT8_C(38), INT8_C(49), -INT8_C(107), INT8_C(57), INT8_C(28), INT8_C(17) },
+      { -INT8_C(7), -INT8_C(32), INT8_C(99), INT8_C(54), INT8_C(32), INT8_C(122), -INT8_C(33), -INT8_C(108) },
+      { INT16_C(775), -INT16_C(20274), -INT16_C(6328), -INT16_C(14199), -INT16_C(18721), -INT16_C(19774), INT16_C(11273), -INT16_C(9682) } },
+    { { INT16_C(27280), -INT16_C(30706), INT16_C(20885), -INT16_C(11166), INT16_C(25965), INT16_C(12772), -INT16_C(23708), INT16_C(4767) },
+      { INT8_C(101), INT8_C(87), -INT8_C(57), -INT8_C(17), INT8_C(109), INT8_C(49), INT8_C(52), INT8_C(15) },
+      { INT8_C(10), -INT8_C(101), -INT8_C(10), INT8_C(37), -INT8_C(38), INT8_C(37), INT8_C(29), INT8_C(64) },
+      { INT16_C(27371), -INT16_C(30518), INT16_C(20932), -INT16_C(11112), INT16_C(26112), INT16_C(12784), -INT16_C(23685), INT16_C(4816) } },
+    { { INT16_C(23152), INT16_C(6589), -INT16_C(15056), INT16_C(20297), -INT16_C(19327), INT16_C(31683), INT16_C(16682), -INT16_C(5678) },
+      { -INT8_C(11), INT8_C(65), INT8_C(24), INT8_C(107), -INT8_C(80), INT8_C(24), -INT8_C(126), -INT8_C(8) },
+      { -INT8_C(56), -INT8_C(110), INT8_C(53), INT8_C(29), -INT8_C(60), INT8_C(121), INT8_C(106), -INT8_C(23) },
+      { INT16_C(23197), INT16_C(6764), -INT16_C(15027), INT16_C(20375), -INT16_C(19307), INT16_C(31780), INT16_C(16914), -INT16_C(5663) } },
+    { { INT16_C(10294), -INT16_C(5817), INT16_C(9100), INT16_C(15766), -INT16_C(19123), -INT16_C(22417), -INT16_C(23716), INT16_C(27840) },
+      { INT8_C(69), -INT8_C(96), -INT8_C(8), -INT8_C(46), INT8_C(7), INT8_C(29), -INT8_C(27), -INT8_C(53) },
+      { INT8_C(7), -INT8_C(102), -INT8_C(104), -INT8_C(70), INT8_C(113), INT8_C(44), INT8_C(29), INT8_C(30) },
+      { INT16_C(10356), -INT16_C(5811), INT16_C(9196), INT16_C(15790), -INT16_C(19017), -INT16_C(22402), -INT16_C(23660), INT16_C(27923) } },
+  };
+
+  for (size_t i = 0 ; i < (sizeof(test_vec) / sizeof(test_vec[0])) ; i++) {
+    simde_int16x8_t a = simde_vld1q_s16(test_vec[i].a);
+    simde_int8x8_t b = simde_vld1_s8(test_vec[i].b);
+    simde_int8x8_t c = simde_vld1_s8(test_vec[i].c);
+    simde_int16x8_t r = simde_vabal_s8(a, b, c);
+
+    simde_test_arm_neon_assert_equal_i16x8(r, simde_vld1q_s16(test_vec[i].r));
+  }
+
+  return 0;
```

Comment:
For future PRs, please include your test matrix generation code; especially for NEON, the test files aren't too big to keep them

Reply:
Ok, I'll remove some test data. I wanted to ask, where should the script for generating test data be placed?

Reply:
No, keep the test data! Also add how you generated it; see https://github.com/simd-everywhere/simde/blob/863239129b17ef1d1b40bcbd9b06db2682ce718d/test/arm/neon/aba.c#L64 for an example

Reply:
I understand. However, I generated the test data using my own script, and I've implemented many instructions (1000+). Will I still need to add the test generation code before sending a pull request later on?

Reply:
My request is just for new PRs, beyond this one.

Reply:
Alright, got it.